lib/migrator.js

```diff
@@ -6,53 +6,158 @@ const { moveFile, removeDirs } = require('./utils/file')
 
 module.exports = class Migrator {
   constructor(options) {
-    this.options = options;
+    const { projectRoot, newComponentStructure } = options;
+
+    this.projectRoot = projectRoot;
+    this.newComponentStructure = newComponentStructure;
   }
 
-  async execute() {
-    let sourceComponentTemplatesPath = path.join(this.options.projectRoot, 'app/templates/components');
-    var sourceComponentTemplateFilePaths = glob.sync(`${sourceComponentTemplatesPath}/**/*.hbs`);
-
-    let sourceComponentPath = path.join(this.options.projectRoot, 'app/components');
-    let sourceComponentFilePaths = glob.sync(`${sourceComponentPath}/**/*.js`);
-    let templatesWithLayoutName = getLayoutNameTemplates(sourceComponentFilePaths);
-    if (templatesWithLayoutName.length) {
-      sourceComponentTemplateFilePaths = sourceComponentTemplateFilePaths.filter(sourceTemplateFilePath => {
-        let sourceTemplatePathInApp = sourceTemplateFilePath.slice(this.options.projectRoot.length); // '/app/templates/components/nested1/nested-component.hbs'
-        let templatePath = sourceTemplatePathInApp.slice('app/templates/'.length); // '/nested1/nested-component.hbs'
-        return !templatesWithLayoutName.includes(templatePath.slice(1).replace('.hbs', ''));
+  findClassicComponentTemplates() {
+    const templateFolderPath = path.join(this.projectRoot, 'app/templates/components');
+    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+
+    return templateFilePaths;
+  }
+
+  findClassicComponentClasses() {
+    const classFolderPath = path.join(this.projectRoot, 'app/components');
+    const classFilePaths = glob.sync(`${classFolderPath}/**/*.js`);
+
+    return classFilePaths;
+  }
+
+  findTemplates() {
+    const templateFolderPath = path.join(this.projectRoot, 'app/templates');
+    const templateFilePaths = glob.sync(`${templateFolderPath}/**/*.hbs`);
+
+    return templateFilePaths;
+  }
+
+  skipTemplatesUsedAsLayoutName(templateFilePaths) {
+    console.info(`\nChecking if any component templates are used as templates of other components using \`layoutName\``);
+
+    const classFilePaths = this.findClassicComponentClasses();
+    const componentsWithLayoutName = getLayoutNameTemplates(classFilePaths);
+
+    if (componentsWithLayoutName.length) {
+      componentsWithLayoutName.sort().forEach(component => {
+        console.info(`❌ Did not move '${component}' due to usage as "layoutName" in a component`);
```

Comment:
We should probably tell them _which_ component

Reply:
Sorry, can you elaborate on what you'd like the log message to say?

I only renamed the value for `forEach` from `template` to `component`:
```javascript
// Before
templatesWithLayoutName.sort().forEach(template => {
  console.info(`❌ Did not move '${template}' due to usage as "layoutName" in a component`);
});

// After
componentsWithLayoutName.sort().forEach(component => {
  console.info(`❌ Did not move '${component}' due to usage as "layoutName" in a component`);
});
```

A test run shows that the log message currently looks like:
```
Checking if any component templates are used as templates of other components using `layoutName`
❌ Did not move 'components/layout-name/layout-name-template' due to usage as "layoutName" in a component
```

Reply:
Ya, I think its fine (didn't notice that it was only a small tweak) for now. I was thinking that saying the template **and** the component that we found using it for `layoutName` would be helpful.